### PR TITLE
Feature/exceptions

### PR DESCRIPTION
**Pull request by prudhomm**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes?
- [x] Have you successfully run the Feel++ testsuite with your changes locally?
- [x] Have you written Doxygen comments in your contribution ?

-----
implements 
- #1779 
- #1780

### CODE REVIEW

File: feelpp/feel/feelcore/exceptions.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                        GitMetadata::authorName(),GitMetadata::authorEmail(),
+                        GitMetadata::commitDate(),
+                        GitMetadata::commitSubject(),
+                        GitMetadata::commitBody() );
+    }
+    else
+    {
+        std::cerr << "WARN: failed to get the current git state. Is this a git repo?" << std::endl;
+    }
+}
```
Comment (vincentchabannes) [inline feelpp/feel/feelcore/exceptions.cpp:90]: maybe call MPI_Abort, else we can have deadlock

**Review by vincentchabannes — APPROVED** [review]

File: feelpp/feel/feelcore/exceptions.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                        GitMetadata::describe(),
+                        GitMetadata::authorName(),GitMetadata::authorEmail(),
+                        GitMetadata::commitDate(),
+                        GitMetadata::commitSubject(),
+                        GitMetadata::commitBody() );
+    }
+    else
+    {
+        std::cerr << "WARN: failed to get the current git state. Is this a git repo?" << std::endl;
+    }
```
Comment (vincentchabannes) [inline feelpp/feel/feelcore/exceptions.cpp:89]: maybe call MPI_Abort, else we can have deadlock

**Review by vincentchabannes — APPROVED** [review]

File: feelpp/feel/feelcore/exceptions.cpp (hunk 2 above)
Comment (prudhomm) [reply feelpp/feel/feelcore/exceptions.cpp:89]: Indeed exception may be thrown by some MPI process, I will check this testcase

Comment (prudhomm) [reply feelpp/feel/feelcore/exceptions.cpp:89]: done

**Pull request merged**